ntoskrnl/io/pnpmgr/pnpnotify.c

```diff
@@ -263,6 +263,7 @@ PiNotifyTargetDeviceChange(
     PAGED_CODE();
 
     PVOID notificationStruct;
+    PTARGET_DEVICE_REMOVAL_NOTIFICATION notifStruct = NULL;
```

Comment:
imo. a simpler fix could be to keep that one where it originally was (in the !IsEqualGUID if block), and then, replace the new `if (notifStruct)` by `if (notificationStruct != CustomNotification)` when you check whether to free the allocated pool pointer.